src/content/blog/posts/iceberg-athena-dbt.md

```diff
@@ -0,0 +1,116 @@
+
+#### Model Generation
+
+Some transformations consist of applying the same logic to multiple tables, for example when deduplicating a dataset made up of multiple tables. This is commonly achieved by looping through a list of tables and applying the same transformation to minimise code redundancy. 
+
+dbt has a one model per table philosophy, which makes it simple and intuitive to use. However, this approach doesn’t support looping, and you can end up creating multiple models with exactly the same logic. Redundant code is difficult to maintain because any changes to the logic will need to be replicated in multiple places. There has been [some discussion](https://github.com/dbt-labs/dbt-core/discussions/5101) to make it possible for dbt to generate models using Jinja, but this is still in progress. 
+
+Instead, we have come up with an approach to generate models using python at run time. We create template `.sql` files to represent each model in a pipeline and generate a model per table. We pass in required parameters, which can be obtained from the glue table properties, or a `.yaml` config file. We have developed integration tests to generate and [build](https://docs.getdbt.com/reference/commands/build) the models and make sure they run as expected.
+
+#### Chunking
+
+Sometimes the demands of a query exceed the resources available to the Athena cluster running the query, or exceeds the query timeout. One solution is to employ chunking. This involves dividing a large dataset into smaller chunks, making each chunk quicker to process and less memory-intensive. Each chunk is then inserted into the final table using the `INSERT INTO` statement. However, this can involve maintaining a lot of boilerplate code. We have developed a [custom](https://docs.getdbt.com/guides/create-new-materializations) insert\_by\_chunk dbt materialization, based on [insert\_by\_period](https://github.com/dbt-labs/dbt-labs-experimental-features/tree/main/insert_by_period), to simplify the chunking process by specifying a single [dbt model config](https://docs.getdbt.com/reference/model-configs). After completing all the `INSERT INTO` operations, we perform various validations to ensure that all the data has been processed accurately. We would like to enhance and release this materialization as open source.
+
+Note that each chunk is processed sequentially (instead of concurrently) due to how dbt handles materializations. However, this limitation is less significant for batch processes, where runtime is less important. Furthermore, chunking is only suitable for queries that can operate on smaller portions of the dataset independently, without requiring access to the entire dataset. Finally, each chunk will incur additional costs as you scan the same data repeatedly, unless the data is partitioned by the same column used for chunking. Again, this limitation is less significant if the initial cost is low.
```

Comment:
"_as you scan the same data repeatedly_" -- switches to first person

Try "_as the same data is repeatedly scanned_"